adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/sasl/AmqpAdapterSaslAuthenticatorFactory.java

```diff
@@ -167,48 +164,24 @@ private void verifyPlain(final String remoteMechanism, final byte[] saslResponse
                 } else {
                     authProvider.authenticate(credentials, completer);
                 }
-                return Future.future();
-            }).recover(t -> {
-                completer.handle(Future.failedFuture(t));
-                return Future.failedFuture(t);
-            });
-
-        }
-
-        private Future<String[]> getFields(final byte[] saslResponse) {
-            final Future<String[]> result = Future.future();
-            final int authzidEndPosition = findNullByte(saslResponse, 0);
-            if (authzidEndPosition < 0) {
-                result.fail(
-                        new CredentialException("PLAIN response is malformed (does not contain an Authorization ID"));
+            } catch (CredentialException e) {
+                completer.handle(Future.failedFuture(e));
             }
-            final int authcidEndPosition = findNullByte(saslResponse, authzidEndPosition + 1);
-            if (authcidEndPosition < 0) {
-                result.fail(new CredentialException("PLAIN response must contain an authentication ID"));
-            }
-            if (authcidEndPosition == saslResponse.length - 1) {
-                result.fail(new CredentialException("PLAIN response must contain a password"));
-            }
-            // Get the authentication identity and password (ignoring authorization id)
-            final String authcid = new String(saslResponse, authzidEndPosition + 1,
-                    authcidEndPosition - (authzidEndPosition + 1), StandardCharsets.UTF_8);
-            final String passwd = new String(saslResponse, authcidEndPosition + 1,
-                    saslResponse.length - (authcidEndPosition + 1));
+        }
 
-            result.complete(new String[] { authcid, passwd });
+    }
 
-            return result;
-        }
+    /**
+     * Represent constants used throughout the Amqp adapter code base.
+     * 
+     * TODO: this class will be moved to a dedicated java file once the list of constants grows.
+     */
+    public final class AmqpAdapterConstants {
```

Comment:
Hmm, why not define this as a class in the main package?
In fact, is it really helpful to have a separate sasl package instead of just the top level package? How many classes are there, five?

Reply:
currently there are 4 classes in total. adding it and the authenticator class will make it 6.

Reply:
so, do you think having them all in one package would be a problem?

Reply:
no not at all, i will update the PR and put them in one package

Reply:
I think that the AMQP_BAD_REQUEST constant might be useful in all services that implement an AMQP server, not only in the AMQP adapter.